modules/fedimint-ln/src/lib.rs

```diff
@@ -856,40 +856,39 @@ impl ServerModule for Lightning {
         vec![
             api_endpoint! {
                 "/account",
-                async |module: &Lightning, dbtx, contract_id: ContractId| -> ContractAccount {
-                    module
-                        .get_contract_account(&mut dbtx, contract_id)
-                        .await
-                        .ok_or_else(|| ApiError::not_found(String::from("Contract not found")))
+                async |module: &Lightning, db, contract_id: ContractId| -> ContractAccount {
+                    Ok(module.await_get_contract_account(db, contract_id).await)
```

Comment:
I don't know if we should hang the connection here waiting for something. We should return `Option<ContractAccount>` and let the user decide if they want to retry, when, how etc.

In the future we might send notifications when new epoch is mineted so users should be able not even retrying until something happened.

If really needed for some reason I would make it a separate call.

Reply:
:shrug: 
I don't know any disadvantage to this. 
It doesn't really "hang" the connection, because json rpc is async. And client can always cancel the request on their side if they need to.

Reply:
>  I don't know any disadvantage to this.

I guess I did not internalize the nature of jsonrpc. Maybe it is a good idea. Ok in that case we might want these calls.

BUT. We have existing calls that do things and expect a response. Are we sure accross the client code and gateway we have no places which are going to get dizzy because of it?

Probably safer to introduce new calls (as we should rename these ones anyway, so the name matches the behavior) and phase out the existing ones in places we did some sanity check which are safe. Also this will help with git bisect etc. in case we did get something wrong.

Reply:
But if you're sure it will be OK, then :+1: and let's rename at some point, probably including getting rid of needless(?) leading slashes from method names.

Reply:
> I don't know any disadvantage to this.

I just though of something. 

It would use less resources on the mint side if clients were waiting for a single "new epoch" notification, instead of (possibly multiple) requests, each getting some spurious wake ups etc.

But we don't have such a notification rn, so :shrug:

Reply:
I think it would be good to have an indication if this will hang or not. Sometimes we want to know that there isn't anything yet. But that will become easier with streams where we can send an initial state, e.g. `None`, that the client can either filter away since it's not the expected thing or use to close the subscription since it actually only wanted the value if currently available.

Reply:
Not going with stream for now. 
Just implementing two types of endpoints: waiting and non waiting version